instrumentation/restlet/restlet-1.0/library/src/main/java/io/opentelemetry/instrumentation/restlet/v1_0/TracingFilter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v1_0;
+
+import static io.opentelemetry.instrumentation.restlet.v1_0.RestletSingletons.instrumenter;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.tracer.ServerSpan;
+import org.restlet.Filter;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+
+public final class TracingFilter extends Filter {
+
+  private static final ThreadLocal<Scope> otelScope = new ThreadLocal<>();
+  private static final ThreadLocal<Context> otelContext = new ThreadLocal<>();
+  private final String path;
+
+  public TracingFilter(String path) {
+    this.path = path;
+  }
+
+  @Override
+  protected int beforeHandle(Request request, Response response) {
+
+    Context parentContext = Context.current();
+
+    if (!instrumenter().shouldStart(parentContext, request)) {
+      return CONTINUE;
```

Comment:
So if there already is a server span in the context the `span.updateName(path)` call won't execute. How about structuring the code a bit differently, e.g.
```java
if (instrumenter().shouldStart(parentContext, request)) {
  // start span
  // start scope
}

Span serverSpan = ServerSpan.fromContextOrNull(currentContext);
if (serverSpan != null) {
  serverSpan.updateName(path);
}
```
This way you'll only create the span when it should be created *and* set the server span name in all cases.

Reply:
Thanks, my approach was indeed incorrect if some server span already existed.